lib/filewatch/since_db.rb

```diff
@@ -0,0 +1,247 @@
+require 'filewatch/boot_setup' unless defined?(FileWatch)
+
+module FileWatch
+  class SinceDb
+    def self.parse_line(line)
+      parts = line.split(" ")
+      parse_line_v2(parts) || parse_line_v1(parts)
+    end
+
+    def self.parse_line_v2(parts)
+      return false if !parts.first.include?(",")
+      fp1 = parts.shift.split(",").map(&:to_i)
+      fp1[0] = Fnv.coerce_bignum(fp1[0])
+      pos = parts.shift.to_i
+      exp = Float(parts.shift)
+      if !parts.empty?
+        fp2 = parts.shift.split(",").map(&:to_i)
+      end
+      [SincedbKey2.new(*fp1), SincedbValue.new(pos, exp, fp2)]
+    end
+
+    def self.parse_line_v1(parts)
+      kparts = parts.shift(3).map(&:to_i)
+      pos, exp = parts
+      exp = Float(exp) if !exp.nil?
+      [SincedbKey1.new(*kparts), SincedbValue.new(pos.to_i, exp)]
+    end
+
+    def self.serialize_kv(k, v)
+      "#{k} #{v}"
+    end
+
+    attr_reader :path, :short_keys
+    attr_writer :logger
+
+    def initialize(opts, loggr)
+      @logger = loggr
+      @path = opts[:sincedb_path]
+      @lastwarn = Hash.new { |h, k| h[k] = 0 }
```

Comment:
I think we already had a discussion about this elsewhere.
Could it be replace with `Hash.new(0)`, I remember it was about some weird edge case.